src/Synth/OscilGen.cpp

```diff
@@ -1078,24 +1244,32 @@ bool OscilGen::needPrepare(OscilGenBuffers& bfrs) const
     return outdated == true || bfrs.oscilprepared == false;
 }
 
+bool OscilGen::mayUseRandom() const
+{
+    return !ADvsPAD && (
+                        Prand > 64 || // phase randomness
+                        Pamprandtype  // amplitude randomness
+                        );
+}
+
 /*
  * Get the oscillator function
+ * When you change this, also change OscilGen::usesRandom
  */
-short int OscilGen::get(OscilGenBuffers& bfrs, float* smps, float freqHz, int resonance) const
+short int OscilGen::get(OscilGenBuffers& bfrs, float* smps, float freqHz, int resonance, float differingBaseFuncPar) const
 {
-    if(needPrepare(bfrs))
-        prepare(bfrs);
+    // note: differingBaseFuncPar can range from 0 to 128, too, but it has steps
+    //       in between to allow better fine tuning
+
+    if(needPrepare(bfrs, differingBaseFuncPar))
+        prepare(bfrs, differingBaseFuncPar);
 
     fft_t *input = freqHz > 0.0f ? bfrs.oscilFFTfreqs.data : bfrs.pendingfreqs;
 
     unsigned int realrnd = prng();
     sprng(randseed);
```

Comment:
While technically outside of the scope of this PR we should eventually switch away from using a rng seed into carrying rng state through various calculations. Right now with multi-threaded operations and a global rng state reproducibility is less of a guarantee. (the single threaded tests are just fine though)

Reply:
Maybe for consistent testing of this PR it might make sense to actually make this in-scope? Right now as it stands it looks like there's roughly 5 locations where the random number generator is getting called. Switching those to prng_r() and carrying around the 'current' RNG state for each of the consumers should be pretty doable. Then there shouldn't be any unexpected randomness introduced in terms of the multi-threaded wavetable generation you have. (This is mostly for benefits in terms of making the code more testable).